yew/src/agent/worker/queue.rs

```diff
@@ -0,0 +1,48 @@
+use std::any::TypeId;
+use std::cell::{RefCell, RefMut};
+use std::collections::{hash_map, HashMap, HashSet};
+
+/// Thread-local instance used to queue worker messages
+pub struct Queue {
+    loaded_agents: RefCell<HashSet<TypeId>>,
+    msgs_queue: RefCell<HashMap<TypeId, Vec<Vec<u8>>>>,
+}
+
+impl Queue {
+    pub fn new() -> Queue {
+        Queue {
+            loaded_agents: RefCell::new(HashSet::new()),
+            msgs_queue: RefCell::new(HashMap::new()),
+        }
+    }
+
+    pub fn borrow_msg_queue_mut(&self) -> RefMut<'_, HashMap<TypeId, Vec<Vec<u8>>>> {
+        self.msgs_queue.borrow_mut()
+    }
+
+    pub fn insert_loaded_agent(&self, type_id: TypeId) {
+        self.loaded_agents.borrow_mut().insert(type_id);
+    }
+
+    pub fn is_worker_loaded(&self, type_id: &TypeId) -> bool {
+        self.loaded_agents.borrow().contains(type_id)
+    }
+
+    pub fn msg_to_queue(&self, msg: Vec<u8>, type_id: TypeId) {
```

Comment:
I think that this naming is clearer.
```suggestion
    pub fn add_msg_to_queue(&self, msg: Vec<u8>, type_id: TypeId) {
```

Reply:
Totally agree!